### PR TITLE
feat: support for WC compatible wallets in react native

### PR DESCRIPTION
**Pull request by joaquim-verges**
## PR-Codex overview
This PR focuses on enhancing wallet management and connection in the thirdweb application for React Native. 

### Detailed summary
- Added `useAutoConnect` hook for autoconnecting wallets
- Updated storage handling for default app metadata
- Improved wallet connection logic in `createWallet`
- Updated dependencies for React Native and `react-native` packages

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/core/hooks/connection/useAutoConnect.ts`, `packages/thirdweb/src/wallets/wallet-connect/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/wallets/wallet-connect/types.ts [hunk 1]
```diff
@@ -181,6 +181,8 @@ export type WCAutoConnectOptions = {
   savedConnectParams?: SavedConnectParams;
 
   chain?: Chain;
+
+  onDisplayUri?: (uri: string) => void;
```
Comment (gregfromstl) [inline packages/thirdweb/src/wallets/wallet-connect/types.ts:185]: Does this need docs for what it is? (not a mobile dev, not sure if its obvious or not)

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 2]
```diff
@@ -258,18 +289,33 @@ async function initProvider(
...
-      if (!isMobile()) {
+      if (!isBrowser()) {
         return;
       }
 
-      const preferUniversal =
-        walletInfo.mobile.universal || walletInfo.mobile.native;
-      const preferNative =
-        walletInfo.mobile.native || walletInfo.mobile.universal;
+      if (!isMobile()) {
```
Comment (gregfromstl) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:316]: If not browser (above if statement) and not mobile continue? What case would get through both of those

File: packages/thirdweb/src/wallets/wallet-connect/types.ts (hunk 1 above)
Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/wallet-connect/types.ts]: I actually dont need this anymore, removing

File: packages/thirdweb/src/wallets/wallet-connect/index.ts (hunk 2 above)
Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/wallet-connect/index.ts:298]: fixed

**Review by gregfromstl — APPROVED** [review]

**Pull request merged**